python/podio/root_io.py

```diff
@@ -10,18 +13,34 @@
 from podio.base_writer import BaseWriterMixin  # pylint: disable=wrong-import-position # noqa: E402
 
 
+def convert_to_str_paths(filenames):
+    """Converts filenames to string paths, handling both string and pathlib.Path objects and
+       iterables of such objects.
+
+    Args:
+        filenames (str, Path, or Iterable[str | Path]): A single filepath or an iterable of
+        filepaths to convert to str object(s).
+
+    Returns:
+        list[str]: A list of filepaths as strings.
+    """
+
+    if isinstance(filenames, Iterable) and not isinstance(filenames, (str, Path)):
+        return [os.fspath(fn) for fn in filenames]
+
+    return [os.fspath(filenames)]
+
+
```

Comment:
I would move this into a separate module that can then be imported by `root_io` and `sio_io` instead of cross-importing this from `root_io` to `sio_io`. Altough I generally don't like `utils` too much, I can't think of a better name at the moment, so I would be OK with that.